examples/hosts/app-integration/external-data/src/view/taskListView.tsx

```diff
@@ -144,6 +151,15 @@ export const TaskListView: React.FC<ITaskListViewProps> = (props: ITaskListViewP
 				setFailedUpdate(true);
 			});
 	}, [taskList]);
+
+	/**
+	 * Set the current client to be the leader of the Fluid document.
+	 */
+	const setLeaderShip = useCallback(() => {
+		console.log("SETTING CLIENT AS LEADER");
+		model.leader = "Leader";
+	}, [model]);
```

Comment:
Props should be considered readonly, so we shouldn't be updating properties inside them. What we probably want here is to receive a function as part of the props, that when called, causes the current client to become the leader; but any change of state that is required in response to that should happen further up in the hierarchy of components.